app/controllers/spree/amazonpay_controller.rb

```diff
@@ -45,6 +45,12 @@ def confirm
     end
 
     amazon_user = SpreeAmazon::User.from_response(body)
+
+    unless amazon_user.valid?
+      redirect_to cart_path, notice: Spree.t(:uid_not_set)
```

Comment:
Optional:

See comment in en.yml to make a change here also to this or similar:
```
Spree.t(:invalid_amazon_user)
```